lib/charms/layer/jenkins/plugins.py

```diff
@@ -11,6 +13,22 @@
 class Plugins(object):
     """Manage Jenkins plugins."""
 
+    def __init__(self):
+        if hookenv.config()["plugins-site"] == "https://updates.jenkins-ci.org/latest/":
+            self.update_center = UpdateCenter()
+        else:
+            plugins_site = hookenv.config()["plugins-site"]
+            try:
+                update_center = plugins_site + "/update-center.json"
+                urllib.request.urlopen(update_center)
+                self.update_center = UpdateCenter(uc_url=update_center)
+            except urllib.error.HTTPError:
+                status_set(
+                    "blocked",
+                    "The configured plugin-site doesn't provide an "
+                    "update-center.json file or is not acessible.")
+                raise
```

Comment:
I think this code needs to decide if it is handling the failure by setting the workload status, or raising an exception for the call site to handle.  If the call sites are handling the exception, it is their job to set the status (feel free to raise PluginError("my message") if you want to pass the error up to the callsite). If the call sites are not handling the exception, then your status_set is wasted as the unit will go into an error state.

Reply:
Addressed in 9bacf3f858ac6b92e4eec4c6bd7d23c5b9cfe869.